docs/the-new-architecture/landing-page.md

```diff
@@ -3,11 +3,26 @@ id: landing-page
 title: Introduction
 ---
 
-This section is the entry point for the new guide’s documentation.
-It contains some basic information about the New Architecture: its pillars, the version from which it has been made available and other very high-level information. Then, it should present the guide’s structure itself.
+Starting in version 0.68, React Native provides a New Architecture that improves several aspects of how JavaScript code and host platforms can integrate, such as:
 
-This section should also contains a sort of Driving guide based on the user use-cases:
+- synchronously measure and render React Native views at the same time as native views
+- support for prioritization of events that update the app's layout, like user interactions
+- improved type safety and consistency of communication between JavaScript and native code
 
-- New user: link to how to use the new [app template](use-app-template)
-- New Library developer: link to the [pillars](pillars)
-- Old library developer/app developer: link to [the migration guide](../new-architecture-intro)
+This is made possible by these [Pillars of the New Architecture](pillars):
+
+- [Fabric renderer and components](pillars-fabric-components), and the capabilities they offer for layout and interaction events
+- [TurboModules](pillars-turbo-modules), which supports faster loading of and communication with native code
+- [CodeGen](pillars-codegen), which generates code you can use to interface with native modules, via static typing in JavaScript
```

Comment:
I'm not very confident in the story we want to tell here. Let me know if there's a better way to frame the pitch for any of this.

I pulled the benefits largely from https://reactnative.dev/architecture/fabric-renderer.

As each named piece we wanted to mention, I did my best to quickly summarize what they provide, but I'm not totally sure about a few things:
* I've mostly read about Fabric as a _renderer_ and the idea of "Fabric components" is new to me. Does that just mean any component that uses a Fabric capability, or any component that's compatible with Fabric?
* I understand the idea and benefits behind JSI, but I'm not sure I'm correctly summarizing what CodeGen does.
* Similarly with TurboModules, I don't understand well yet everything the concept contains. I based this summary off the understanding I got from previously reading https://formidable.com/blog/2019/fabric-turbomodules-part-3/.

Reply:
[Here](https://github.com/facebook/react-native-website/pull/3046) I wrote about the main pillars.
Fabric is the name of the Renderer. A Fabric Component (FC) is a UI component that leverage the Fabric Renderer.
TurboModules (TM) are libraries that don't have UI parts and that leverages the JSI mechanism to communicate between JavaScript and the platform code.

CodeGen is a meta-pillar: we introduced that section in the Pillars because it is important that we explain what it does and ow it works after the reader has some concepts about the TM and FC development cycle. It is not a real pillar because, ideally, it must stay hidden from the users' eyes. However, we need to document it because they have to write code keeping the generated code into consideration.

I hope that this short explanation helps you with these concepts. Feel free to ask more questions or we can have a meeting if that could speed you up.

Reply:
> A Fabric Component (FC) is a UI component that leverage the Fabric Renderer.

What is meant by "leverage" is still ambiguous to me: does that mean that the component is:
a) simply compatible with the Fabric renderer (e.g., my custom component that just renders a customized `<View>` would be a Fabric component, since `View` is compatible with Fabric), or 
b) that it implements code that directly interfaces with native views (e.g., a "host component" as defined in https://reactnative.dev/architecture/glossary#react-host-components-or-host-components), or 
c) that it specifically utilizes capabilities that are unique to Fabric?

> TurboModules (TM) are libraries that don't have UI parts

That is an interesting distinction, I don't think I understood that before. Why do they specifically not have UI parts? Because that would be considered a Fabric component?

On CodeGen, I think I feel less confident in describing it because I don't know the concrete details of what is actually being generated and how that generated code is used. I think in part I need to try it out.

Reply:
> What is meant by "leverage" is still ambiguous to me: does that mean that the component is:

RN with the New Architecture, we have two different renderers: Paper (the old architecture) and Fabric (the new architecture). A renderer is a piece of code that draws elements on the screens.
* Native Components (from the old architecture) use Paper to be drawn on the screen.
* Fabric Components use Fabric. 

I think that this is the easiest way to describe the difference. Technically, to actually use Farbic, a Fabric Component has to implement a C++ interface (whose boilerplate is automatically generated) and it has to be registered in a special component registry (done automatically for iOS, it requires a little bit of manual steps for Android).

> That is an interesting distinction, I don't think I understood that before. Why do they specifically not have UI parts? Because that would be considered a Fabric component?

TL;DR: yes. 
The long answer is that Fabric is tightly bound to the UI and how we draw elements on the screen. However, not all the libraries requires to draw stuff on the screen (imagine a Networking library, for example). In those case, it does not make sense to download and install all the UI related dependencies. To allow this optimization, we split TM and FC in two different pillars. 

> On CodeGen, I think I feel less confident in describing it because I don't know the concrete details of what is actually being generated and how that generated code is used. I think in part I need to try it out.

That make sense. However, for this page, I think that it's enough to specify that the codeGen is an automatic steps that pregenerate a lot of C++ boilerplate code that is required by the new architecture. Ideally, devs should never see the automatically generated code. But it is not like that at the moment, so we need to at least cite it.

Reply:
Alright, I tried rephrasing to what I think adds more relevant but broad context. Let me know whether you think it's an improvement.